applications: nrf_desktop: remove spare configurations (#7385)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * Documentation for selective HID report subscription in :ref:`nrf_desktop_usb_state` using :kconfig:option:`CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION` option.

* Removed:

>   * Configurations without bootloader. By default all boards have B0 bootloader enabled if it fits the memory.

Reviewer: Isn't it only for the one mentioned in the nRFdesktop_readme?

Author: No, the change I do is about making configuration with b0 default configuration

>   * Configurations without bootloader. By default all boards have B0 bootloader enabled if it fits the memory.

Reviewer: Isn't it only for the one mentioned in the nRFdesktop_readme?

> * Removed:
> 
>   * Configurations without bootloader. By default all boards have B0 bootloader enabled if it fits the memory.

Reviewer: ```suggestion
* Removed configurations without a bootloader. 
  The B0 bootloader is enabled by default on all boards if the configuration fits in memory.
```

  * Documentation for selective HID report subscription in :ref:`nrf_desktop_usb_state` using :kconfig:option:`CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION` option.

* Removed configurations without a bootloader.

>   The B0 bootloader is enabled by default on all boards if the configuration fits in memory.

Reviewer: ```suggestion
  The B0 bootloader is enabled by default on all boards if the configuration with two image slots fits in memory.
  Alternatively, MCUboot bootloader with a single image slot and serial recovery is enabled.
  In case the configuration with the MCUboot does not fit in memory, no bootloader is enabled.
```
